application/models/Question.php

```diff
@@ -778,23 +778,37 @@ private function sortAnswerOptions($answerOptions, $language = null)
             }
             return $answerOptions;
         }
-
         // Sort alphabetically if applicable
         if ($this->shouldOrderAnswersAlphabetically()) {
             if (empty($language) || !in_array($language, $this->survey->allLanguages)) {
                 $language = $this->survey->language;
             }
+            $locale = $language;
+            $aLanguageData = getLanguageData();
+            if (isset($aLanguageData[$language]['cldr'])) {
+                $locale = $aLanguageData[$language]['cldr'];
+            }
+            // need php_intl
+            if (function_exists('collator_create')) {
```

Comment:
Can we create a wrapper for the collator? or maybe for the sorting procedures?
Then we wouldn't be doing function_exists

Reply:
> Can we create a wrapper for the collator? or maybe for the sorting procedures? Then we wouldn't be repeating function_exists so much

You mean 2 different functions with 80% same code ? Unsure on the best option here.

Reply:
I meant a function like this:
```
collator_sort($sorted, $collator_sort_type, $fallback_sort_type) {
            // need php_intl
            if (function_exists('collator_create')) {
                $collator = collator_create($locale);
                collator_asort($collator, $sorted, collator_sort_type);
            } else {
                Yii::log("For better alphabeticall ordering : you need php-intl.", 'warning');
                asort($sorted, $fallback_sort_type);
            }            
}
```

Reply:
Inside a foreach : creating a new Collator for each foreach ?

Reply:
If i move too just use an uasort inside the foreach.

Reply:
> Inside a foreach : creating a new Collator for each foreach ?

Yep, you are right. There other ways, like creating a class.
`class CollatorHelper` with a sort method.
But maybe is not needed. Leave it up to you.

Reply:
It's a foreach for scale : maybe we can do it ? Since we have maximum 2 scale ?